pkg/dataloader/jiraloader/jiraloader.go

```diff
@@ -0,0 +1,331 @@
+					resolutionTime = &createdTime
+				}
+			}
+
+			// OCPBUGS don't get a resolution time until it's closed which happens when a release GA's. We
+			// find the first terminal incident status changelog instead. From TRT's perspective, we don't
+			// care about OCPBUGS incidents after they go to MODIFIED.
+			resolvedStatuses := []string{"MODIFIED", "ON_QA", "Verified", "Closed"}
+			for _, status := range resolvedStatuses {
+				if item.ToString == status {
+					createdTime, err := time.Parse(changelogLayout, history.Created)
+					if err != nil {
+						log.WithError(err).Warningf("parsing error: %s", history.Created)
+						continue
+					}
+					// We pick the oldest state change
+					if resolutionTime == nil || resolutionTime.After(createdTime) {
+						log.Debugf("%s to %s at %+v", issue.Key, status, createdTime)
+						resolutionTime = &createdTime
+					}
+				}
+			}
+		}
+	}
+
+	// Fallback to the jira resolution time
+	if issue.Fields.ResolutionDate != "" && resolutionTime == nil {
+		jiraResolutionTime, err := time.Parse(jiraTimeLayout, issue.Fields.ResolutionDate)
+		if err != nil {
+			fmt.Printf("parsing error: %+v", err)
+		}
+		log.Debugf("resolution time for %s is %+v", issue.Key, jiraResolutionTime)
+		resolutionTime = &jiraResolutionTime
+	}
+
+	return resolutionTime
+}
+
+func issueContainsLabel(issue *v1jira.Issue, label string) bool {
+	for _, issueLabel := range issue.Fields.Labels {
+		if issueLabel == label {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (jl *JiraLoader) componentLoader() {
+	start := time.Now()
+	log.Infof("loading jira ocpbugs component information...")
+	body, err := jiraRequest("https://issues.redhat.com/rest/api/2/project/12332330/components")
+	if err != nil {
+		jl.errors = append(jl.errors, err)
+		return
+	}
+
+	var components []v1jira.Component
+	err = json.Unmarshal(body, &components)
+	if err != nil {
+		jl.errors = append(jl.errors, err)
+		return
+	}
+
+	var ids []uint
+	for _, c := range components {
+		jiraID, err := strconv.ParseUint(c.ID, 10, 64)
+		if err != nil {
+			msg := "error parsing jira ID"
+			log.WithError(err).Warn(msg)
+			jl.errors = append(jl.errors, errors.WithMessage(err, msg))
+			continue
+		}
+
+		mc := models.JiraComponent{
+			Model: models.Model{
+				ID: uint(jiraID),
+			},
+			Name:        c.Name,
+			Description: c.Description,
+			LeadName:    c.Lead.DisplayName,
+			LeadEmail:   c.Lead.Name,
+		}
+
+		if err := jl.dbc.DB.Clauses(clause.OnConflict{UpdateAll: true}).Save(&mc).Error; err != nil {
+			jl.errors = append(jl.errors, err)
+			log.WithError(err).Warningf("failed to save component %q", c.Name)
+			continue
+		}
+		ids = append(ids, mc.ID)
+	}
+
+	log.Infof("deleting old records...")
+	oldRecords := jl.dbc.DB.Where("id NOT IN ?", ids).Unscoped().Delete(&models.JiraComponent{})
+	if oldRecords.Error != nil {
+		log.WithError(oldRecords.Error).Warningf("couldn't delete old records")
+		jl.errors = append(jl.errors, oldRecords.Error)
+	}
```

Comment:
Totally wasn't necessary to switch to another variant I just wanted to learn the benefits as it's not an approach I've used before. I'm ok with either you want to proceed with.

Reply:
No this is fine, I've gone back and forth on this a bunch and AFAIK we're not close to any limits for the tables we're doing now.